nf_core/lint/files_unchanged.py

```diff
@@ -107,7 +107,9 @@ def files_unchanged(self):
     logging.getLogger("nf_core.create").setLevel(logging.ERROR)
 
     # Generate a new pipeline with nf-core create that we can compare to
-    test_pipeline_dir = os.path.join(tempfile.mkdtemp(), "nf-core-{}".format(short_name))
+    tmp_dir = tempfile.mkdtemp()
```

Comment:
This was the only change made outside `tests` directory

Reply:
I don't really understand how splitting these two lines up makes any functional difference?

Reply:
well `tempfile.mkdtemp()` create a temporary directory like `/tmp/tmp-abcdfe`, then you append the `nf-core-{short_name}` to the path. `test_pipeline_dir` will be equal to `/tmp/tmp-abcdfe/nf-core-{short-name}`, so if I remove such dir the `/tmp/tmp-abcdfe` will remain. I track this filename in a different variable to delete it (and its content) after in code. Without doing this, I need to process the parent folder, for example:
```python
from pathlib import Path
parent_dir = Path(test_pipeline_dir).parent
shutil.rmtree(parent_dir)
```